packages/augur-core/src/contracts/trading/ERC20Proxy.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity 0.5.15;
+
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract ERC20ProxyNexus {
+    using SafeMathint256 for uint256;
+
+    address public target1155;
+    address proxyToClone;
+
+    // tokenId -> owner -> spender -> allowance
+    // The spender is whoever called the proxy contract, which is msg.sender in the proxy and _caller here in the nexus.
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
+    // tokenId -> proxy contract
+    mapping (uint256 => address) public proxies;
+
+    constructor(address _proxyToClone, address _target1155) public {
+        proxyToClone = _proxyToClone;
+        target1155 = _target1155;
+        // This contract will be able to do anything with the user's 1155 tokens.
+        IERC1155(target1155).setApprovalForAll(address(this), true);
+    }
+
+    function newERC20(uint256 _tokenId) public returns (ERC20Proxy) {
+        require(proxies[_tokenId] == address(0)); // must not have been registered yet
+        _erc20 = ERC20Proxy(createClone(proxyToClone));
+        proxies[_tokenId] = _erc20; // register
+        _erc20.initialize(this, _tokenId);
+        return _erc20;
+    }
+
+    function transfer(address _proxy, address _caller, address _to, uint256 _amount) public returns (bool) {
```

Comment:
This and the methods below need to validate that the msg.sender is a registered ERC20 proxy

Reply:
Ah I see, this is done via the `_gate` calls in the internal methods. NM